### ISSUE TITLE
May 2022 Endgame

### ISSUE THREAD

**Issue by joaomoreno** [microsoft/vscode#150378]
- *05/30* Code freeze for the endgame
- *06/03* [Endgame done](https://github.com/Microsoft/vscode/wiki/Development-Process#inside-an-iteration)
- *06/09* Expected release date (this may change)

##### Monday
- [x] Check that all queries in this issue use the current milestone @joaomoreno
- [x] Run [OSS tool](https://github.com/microsoft/vscode-build-tools/tree/main/distro-tools) @joaomoreno
- [x] Update links in the [Endgame notebooks](https://github.com/microsoft/vscode/blob/main/.vscode/notebooks/endgame.github-issues) to point to new milestone @joaomoreno
- [x] Code freeze at 5pm PT
- [x] Ensure we have a green build on all platforms at 5pm PT
- [x] [🔖All closed feature-requests](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed+milestone%3A%22May+2022%22+label%3Afeature-request+-label%3Averification-needed+-label%3Aon-testplan+-label%3Averified+-label%3A*duplicate+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-dev+repo%3Amicrosoft%2Fvscode-remote-repositories-github+repo%3Amicrosoft%2Fvscode-unpkg+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github+repo%3Amicrosoft%2Fvscode-livepreview+repo%3Amicrosoft%2Fvscode-python+repo%3Amicrosoft%2Fvscode-jupyter+repo%3Amicrosoft%2Fvscode-jupyter-internal+repo%3Amicrosoft%2Fvscode-settings-sync-server+repo%3Amicrosoft%2Fvscode-emmet-helper+repo%3Amicrosoft%2Fvscode-remotehub) either have a `verification-needed` or `on-testplan` label
- [x] Create test plan items following the template [here](https://github.com/microsoft/vscode/wiki/Writing-Test-Plan-Items) by 6pm PT
- [x] Update your availability for testing here - https://vscode-tools.azurewebsites.net/team-manifest @microsoft/vscode
  - [x] Update availability of testers in vacation. Double check N/A testers. @joaomoreno

##### Tuesday
- [x] Test plan items assigned (using https://vscode-tools.azurewebsites.net/test-plan-items)
  - Run the tool multiple times to balance load if test items come in later and assignments are already made
  - [Assigned to you](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+assignee%3A%40me++label%3Atestplan-item+)
- [x] Test build starts at 7am CET
- [x] Test plan ready by 8am CET
- [x] [🔖Testing](https://github.com/issues?q=is%3Aopen+is%3Aissue+label%3Atestplan-item+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-dev+repo%3Amicrosoft%2Fvscode-remote-repositories-github+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github+repo%3Amicrosoft%2Fvscode-livepreview+repo%3Amicrosoft%2Fvscode-python+repo%3Amicrosoft%2Fvscode-jupyter+repo%3Amicrosoft%2Fvscode-jupyter-internal+milestone%3A%22May+2022%22)
- [x] [🔖Verification needed](https://github.com/issues?q=is%3Aissue+milestone%3A%22May+2022%22+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-dev+repo%3Amicrosoft%2Fvscode-remote-repositories-github+repo%3Amicrosoft%2Fvscode-unpkg+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github+repo%3Amicrosoft%2Fvscode-livepreview+repo%3Amicrosoft%2Fvscode-python+repo%3Amicrosoft%2Fvscode-jupyter+repo%3Amicrosoft%2Fvscode-jupyter-internal+repo%3Amicrosoft%2Fvscode-settings-sync-server+repo%3Amicrosoft%2Fvscode-emmet-helper+repo%3Amicrosoft%2Fvscode-remotehub+-label%3Aiteration-plan+-label%3Aendgame-plan+-label%3Atestplan-item+label%3Averification-needed+-label%3Averified+is%3Aclosed+label%3Afeature-request)

##### Wednesday
- [x] Remind team members to assign issues that they intend to fix to the current milestone
- [x] [🔖Testing](https://github.com/issues?q=is%3Aopen+is%3Aissue+label%3Atestplan-item+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-dev+repo%3Amicrosoft%2Fvscode-remote-repositories-github+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github+repo%3Amicrosoft%2Fvscode-livepreview+repo%3Amicrosoft%2Fvscode-python+repo%3Amicrosoft%2Fvscode-jupyter+repo%3Amicrosoft%2Fvscode-jupyter-internal)
- [x] Fixing (self-assigned, milestone assigned)
- [x] [🔖Verification needed](https://github.com/issues?q=is%3Aissue+milestone%3A%22May+2022%22+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-dev+repo%3Amicrosoft%2Fvscode-remote-repositories-github+repo%3Amicrosoft%2Fvscode-unpkg+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github+repo%3Amicrosoft%2Fvscode-livepreview+repo%3Amicrosoft%2Fvscode-python+repo%3Amicrosoft%2Fvscode-jupyter+repo%3Amicrosoft%2Fvscode-jupyter-internal+repo%3Amicrosoft%2Fvscode-settings-sync-server+repo%3Amicrosoft%2Fvscode-emmet-helper+repo%3Amicrosoft%2Fvscode-remotehub+-label%3Aiteration-plan+-label%3Aendgame-plan+-label%3Atestplan-item+label%3Averification-needed+-label%3Averified+is%3Aclosed+label%3Afeature-request)

##### Thursday
- [x] Fixing (self-assigned, milestone assigned, no need for PR or review)
  - Increased scrutiny sets in due to testing being completed. Fixes pose a much higher risk
  - Move issues to the next month that can be deferred
- [x] [🔖Verification needed](https://github.com/issues?q=is%3Aissue+milestone%3A%22May+2022%22+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-dev+repo%3Amicrosoft%2Fvscode-remote-repositories-github+repo%3Amicrosoft%2Fvscode-unpkg+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github+repo%3Amicrosoft%2Fvscode-livepreview+repo%3Amicrosoft%2Fvscode-python+repo%3Amicrosoft%2Fvscode-jupyter+repo%3Amicrosoft%2Fvscode-jupyter-internal+repo%3Amicrosoft%2Fvscode-settings-sync-server+repo%3Amicrosoft%2Fvscode-emmet-helper+repo%3Amicrosoft%2Fvscode-remotehub+-label%3Aiteration-plan+-label%3Aendgame-plan+-label%3Atestplan-item+label%3Averification-needed+-label%3Averified+is%3Aclosed+label%3Afeature-request)
- [x] [🔖Verification](https://github.com/issues?q=is%3Aissue+milestone%3A%22May+2022%22+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-dev+repo%3Amicrosoft%2Fvscode-remote-repositories-github+repo%3Amicrosoft%2Fvscode-unpkg+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github+repo%3Amicrosoft%2Fvscode-livepreview+repo%3Amicrosoft%2Fvscode-python+repo%3Amicrosoft%2Fvscode-jupyter+repo%3Amicrosoft%2Fvscode-jupyter-internal+repo%3Amicrosoft%2Fvscode-settings-sync-server+repo%3Amicrosoft%2Fvscode-emmet-helper+repo%3Amicrosoft%2Fvscode-remotehub+-label%3Aiteration-plan+-label%3Aendgame-plan+-label%3Atestplan-item+label%3Abug+-label%3Aduplicate+-label%3A*duplicate+-label%3Ainvalid+-label%3Aas-designed+-label%3Aerror-telemetry+is%3Aclosed+-label%3Averified)

##### Friday
- [x] Build a `stable` build to ensure stable build is green @joyceerhl 
- [x] Pause scheduled `insider` builds @joyceerhl 
- Satellite modules/npm packages ready, version updated, smoke tested
  - [x] yo generator **@aeschli**
  - [x] js-debug **@connor4312**
  - [x] node debugadapter node **@weinand**
  - [x] debug adapter protocol **@weinand**
  - [x] custom data for html/css **@aeschli**
- [x] Only candidate issues are open and assigned to [🔖milestone](https://github.com/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22May+2022%22+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-dev+repo%3Amicrosoft%2Fvscode-remote-repositories-github+repo%3Amicrosoft%2Fvscode-unpkg+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github+repo%3Amicrosoft%2Fvscode-livepreview+repo%3Amicrosoft%2Fvscode-python+repo%3Amicrosoft%2Fvscode-jupyter+repo%3Amicrosoft%2Fvscode-jupyter-internal+repo%3Amicrosoft%2Fvscode-settings-sync-server+repo%3Amicrosoft%2Fvscode-emmet-helper+repo%3Amicrosoft%2Fvscode-remotehub+)
- [x] All issues [🔖verified](https://github.com/issues?q=is%3Aissue+milestone%3A%22May+2022%22+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-dev+repo%3Amicrosoft%2Fvscode-remote-repositories-github+repo%3Amicrosoft%2Fvscode-unpkg+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github+repo%3Amicrosoft%2Fvscode-livepreview+repo%3Amicrosoft%2Fvscode-python+repo%3Amicrosoft%2Fvscode-jupyter+repo%3Amicrosoft%2Fvscode-jupyter-internal+repo%3Amicrosoft%2Fvscode-settings-sync-server+repo%3Amicrosoft%2Fvscode-emmet-helper+repo%3Amicrosoft%2Fvscode-remotehub+-label%3Aiteration-plan+-label%3Aendgame-plan+-label%3Atestplan-item+label%3Abug+-label%3Aduplicate+-label%3A*duplicate+-label%3Ainvalid+-label%3Aas-designed+-label%3Aerror-telemetry+is%3Aclosed+-label%3Averified)
- [x] All open PRs on the milestone [🔖merged or deferred](https://github.com/issues?q=is%3Apr+milestone%3A%22May+2022%22+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-dev+repo%3Amicrosoft%2Fvscode-remote-repositories-github+repo%3Amicrosoft%2Fvscode-unpkg+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github+repo%3Amicrosoft%2Fvscode-livepreview+repo%3Amicrosoft%2Fvscode-python+repo%3Amicrosoft%2Fvscode-jupyter+repo%3Amicrosoft%2Fvscode-jupyter-internal+repo%3Amicrosoft%2Fvscode-settings-sync-server+repo%3Amicrosoft%2Fvscode-emmet-helper+repo%3Amicrosoft%2Fvscode-remotehub+is%3Aopen)
- [x] Branch code to `release/<x.y>` after all expected fixes are in (latest 5PM PST) @joyceerhl 
- [x] Branch distro to `release/<x.y>` after all expected fixes are in (latest 5PM PST) @joyceerhl 
- [x] Announce `main` is open for business @joyceerhl 
- [x] Fixing (PR + review required once branched - major bugs only - to be discussed in stand-up meeting, labeled as `candidate`)
- [x] All release notes updated
  - release notes are collected in a file named *`v<Major>_<Minor>.md`* in this [repo directory](https://github.com/Microsoft/vscode-docs/blob/vnext/release-notes/)
  - [ ] @aeschli
  - [x] @alexdima
  - [x] @alexr00
  - [ ] @bpasero
  - [x] @chrmarti
  - [x] @connor4312
  - [x] @dbaeumer
  - [x] @deepak1556
  - [x] @egamma
  - [ ] @hediet
  - [x] @isidorn
  - [ ] @JacksonKearl
  - [x] @joaomoreno
  - [x] @joyceerhl
  - [x] @jrieken
  - [ ] @kieferrm
  - [ ] @lramos15
  - [x] @lszomoru
  - [ ] @meganrogge
  - [ ] @mjbvz
  - [ ] @rebornix
  - [x] @roblourens
  - [x] @rzhao271
  - [x] @sandy081
  - [ ] @sbatten
  - [x] @tanhakabir
  - [ ] @tylerleonhardt
  - [ ] @tyriar
  - [x] @weinand
- [x] Acknowledge pull requests in release notes. We acknowledge PRs from outside the team. We have improved [the tooling](https://vscode-tools.azurewebsites.net/acknowledgement) so that the endgame champion can generate the pull request acknowledgment for all repositories at once. @joaomoreno
  - [x] `debug-adapter-protocol`, `inno-updater`, `jsonc-parser`, `language-server-protocol`, `lsif-node`, `vscode`, `vscode-codicons`, `vscode-css-languageservice`, `vscode-debugadapter-node`, `vscode-dev-containers`, `vscode-docs`, `vscode-emmet-helper`, `vscode-eslint`, `vscode-extension-samples`, `vscode-generator-code`, `vscode-hexeditor`, `vscode-html-languageservice`, `vscode-js-debug`, `vscode-js-debug-companion`, `vscode-js-profile-visualizer`, `vscode-jshint`, `vscode-json-languageservice`, `vscode-languageserver-node`, `vscode-livepreview`, `vscode-loader`, `vscode-lsif-extension`, `vscode-node-debug`, `vscode-node-debug2`, `vscode-pull-request-github`, `vscode-recipes`, `vscode-references-view`, `vscode-textmate`, `vscode-vsce`
- [x] Acknowledge [issue trackers](https://github.com/microsoft/vscode-internalbacklog/wiki/Community-Triage---Credits) from the community **@chrmarti**
- [x] Add notable fixes to the release notes **all**
- When done fixing/verifying and there are changes since last build at the end of day PT
  - [x] Build and manually release Insider from release/<x.y> @joyceerhl
- [x] Localization: Run [Update VS Code Branch](https://github.com/microsoft/vscode-loc-drop/actions/workflows/update-vscode-branch.yml) in the vscode-loc-drop repo with `release/*` as the VS Code Branch parameter (it's the default so you shouldn't have to change anything) @joyceerhl

##### Friday/Monday
- [x] Polish release notes **redmond**
- [x] Fixing (only critical bugs - no string changes)

##### Monday - Wednesday
- [x] [🔖milestone issues](https://github.com/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22May+2022%22+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-dev+repo%3Amicrosoft%2Fvscode-remote-repositories-github+repo%3Amicrosoft%2Fvscode-unpkg+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github+repo%3Amicrosoft%2Fvscode-livepreview+repo%3Amicrosoft%2Fvscode-python+repo%3Amicrosoft%2Fvscode-jupyter+repo%3Amicrosoft%2Fvscode-jupyter-internal+repo%3Amicrosoft%2Fvscode-settings-sync-server+repo%3Amicrosoft%2Fvscode-emmet-helper+repo%3Amicrosoft%2Fvscode-remotehub+)
- [x] [🔖candidate issues](https://github.com/issues?q=is%3Aissue+label%3Acandidate+repo%3Amicrosoft%2Fvscode+repo%3Amicrosoft%2Fvscode-internalbacklog+repo%3Amicrosoft%2Fvscode-dev+repo%3Amicrosoft%2Fvscode-remote-repositories-github+repo%3Amicrosoft%2Fvscode-unpkg+repo%3Amicrosoft%2Fvscode-remote-release+repo%3Amicrosoft%2Fvscode-js-debug+repo%3Amicrosoft%2Fvscode-pull-request-github+repo%3Amicrosoft%2Fvscode-livepreview+repo%3Amicrosoft%2Fvscode-python+repo%3Amicrosoft%2Fvscode-jupyter+repo%3Amicrosoft%2Fvscode-jupyter-internal+repo%3Amicrosoft%2Fvscode-settings-sync-server+repo%3Amicrosoft%2Fvscode-emmet-helper+repo%3Amicrosoft%2Fvscode-remotehub+milestone%3A%22May+2022%22)
- [x] Polish release notes **redmond**
- [x] Cherry-pick hand-picked and reviewed changes to `release/<x.y>` @joaomoreno
- [x] Build `Insider` from `release/<x.y>` @joaomoreno
- [x] Manually release `Insider` @joaomoreno
- [x] Build stable for all platforms as new candidate issues come in @joaomoreno
- [x] Documentation updated
  - [ ] @aeschli
  - [x] @alexdima
  - [ ] @alexr00
  - [ ] @bpasero
  - [x] @chrmarti
  - [x] @connor4312
  - [x] @dbaeumer
  - [x] @deepak1556
  - [x] @egamma
  - [ ] @hediet
  - [x] @isidorn
  - [ ] @JacksonKearl
  - [x] @joaomoreno
  - [ ] @joyceerhl
  - [ ] @jrieken
  - [ ] @kieferrm
  - [ ] @lramos15
  - [x] @lszomoru
  - [ ] @meganrogge
  - [ ] @mjbvz
  - [ ] @rebornix
  - [x] @roblourens
  - [x] @rzhao271
  - [x] @sandy081
  - [ ] @sbatten
  - [ ] @tylerleonhardt
  - [ ] @tyriar
  - [ ] @weinand
- [x] Run `scripts/test-documentation.sh|bat` and add file or fix issues if there are new colors that are not documented. @joaomoreno

> **Note:** The `Insiders` build needs to be in the wild for 24 hours before we can enter the last phase of the endgame. @joaomoreno

##### Wednesday/Thursday - Expected release day (this may change)
- [x] Build stable for all platforms @joaomoreno
- [x] Sanity check of installable bits ([server instructions](https://github.com/microsoft/vscode-remote-release/wiki/Sanity-Check-VS-Code-Servers))
  - [x] Windows 32 bit @hediet 
    - [x] signed installer 32-bit
    - [x] signed user installer 32-bit
    - [x] zip 32-bit
    - [x] server 32-bit
  - [x] Windows 64 bit @dbaeumer 
    - [x] signed installer 64-bit
    - [x] signed user installer 64-bit
    - [x] zip 64-bit
    - [x] server 64-bit
  - [x] Windows ARM64 @joaomoreno 
    - [x] signed installer ARM64
    - [x] signed user installer ARM64
    - [x] zip ARM64
  - [x] macOS
    - [x] Universal @jrieken 
    - [x] Intel @connor4312 
    - [x] Intel server @connor4312 
    - [x] Apple Silicon @jrieken 
  - [x] Linux x64 @aeschli 
    - [x] deb
    - [x] rpm
    - [x] archives
    - [x] snap (`sudo snap install --classic --dangerous <file>.snap`)
  - [x] Linux server @chrmarti 
    - [x] x64
    - [x] x64 Alpine
    - [x] ARM32
    - [x] ARM64
- [x] Publish website **@gregvanl**
- [x] Publish to stable @joyceerhl
- [x] Trigger the [vscode.dev Deploy pipeline](https://dev.azure.com/monacotools/Monaco/_build?definitionId=276) for the `prod` deployment target @joyceerhl
- [x] Create an official release @joyceerhl
  - [x] Create a tag (make sure you pull the release branch first): `git tag <x.y.z>`
  - [x] Push the tag: `git push origin <x.y.z>`
  - [x] Create a GitHub release: [Open the GitHub tags](https://github.com/microsoft/vscode/tags), and click far right `... > Create Release`. Use the correct title and description from our release notes. Also change the relative links for the key highlight list items to absolute links [Example](https://github.com/microsoft/vscode/releases/tag/1.39.0)
- [x] Twitter announcement **@chrisdias**
- [x] Enable scheduled `insider` builds @joyceerhl
- [x] Bump up the version in `package.json` on `main` - @joyceerhl
- [x] [Publish @types/vscode](https://github.com/microsoft/vscode/wiki/Publish-vscode-types) @joyceerhl
- [x] Close the milestone on [GitHub](https://github.com/microsoft/vscode/milestones) @joyceerhl
- [x] Create next milestone and ensure that it has a due date @joyceerhl
- [x] If a [recovery build](https://github.com/microsoft/vscode/wiki/Running-the-Endgame#recovery-build) is needed, proceed to https://github.com/microsoft/vscode/wiki/Endgame-Recovery-Template or https://github.com/microsoft/vscode-internalbacklog/wiki/Patch-Tuesday-Release-Endgame-Template/ @joyceerhl
- [x] If no Patch Tuesday release is planned, [let the Updates team know of that fact](https://github.com/microsoft/vscode-internalbacklog/wiki/Patch-Tuesday-E-Mail-Template) ([CSEClientPatching@microsoft.com](mailto:CSEClientPatching@microsoft.com)) @joyceerhl
